duecredit/io.py

```diff
@@ -190,10 +190,18 @@ def dump(self, tags=None):
             self.fd.write('\n')
 
 def get_text_rendering(citation, style='harvard1'):
+    from .collector import Citation
     entry = citation.entry
     if isinstance(entry, Doi):
         bibtex_rendering = get_bibtex_rendering(entry)
-        return get_text_rendering(bibtex_rendering)
+        # pass down the kwargs
+        bibtex_citation = Citation(bibtex_rendering,
+                                   description=citation.description,
+                                   path=citation.path,
+                                   version=citation.version,
+                                   cite_module=citation.cite_module,
+                                   tags=citation.tags)
```

Comment:
future todo -- I don't like that this 'cloning' happens outside of the Citation code.  We should may be make `entry` a property with a setter, and then just copy old citation (copy.copy) into a new one and override `entry` with bibtex one